echopype/calibrate/calibrate_base.py

```diff
@@ -7,13 +13,158 @@
     "AZFP": ("EL", "DS", "TVR", "VTX", "equivalent_beam_angle", "Sv_offset"),
 }
 
+DataKind = Literal["stationary", "mobile", "organized"]
+InterpMethod = Literal["linear", "nearest", "zero", "slinear", "quadratic", "cubic"]
+VALID_INTERP_METHODS: Dict[DataKind, List[InterpMethod]] = {
+    "stationary": ["linear", "nearest", "zero", "slinear", "quadratic", "cubic"],
+    "mobile": ["linear", "nearest", "cubic"],
+    "organized": ["linear", "nearest"],
+}
+
+
+class EnvParams:
+    def __init__(
+        self,
+        env_params,
+        data_kind: DataKind,
+        interp_method: InterpMethod = "linear",
+        extrap_method: Optional[Literal["linear", "nearest"]] = None,
+    ):
+        """
+        Interp methods limited depending on data_kind
+        Extrap only available for stationary and organized
+        """
+        if interp_method not in VALID_INTERP_METHODS[data_kind]:
+            raise ValueError(
+                f"invalid interp_method {interp_method} for data_kind {data_kind}"
+            )
+
+        self.env_params = env_params
+        self.data_kind = data_kind
+        self.interp_method = interp_method
+        self.extrap_method = extrap_method
+
+    def _apply(self, echodata) -> Dict[str, xr.DataArray]:
+        if self.data_kind == "stationary":
+            dims = ["ping_time"]
+        elif self.data_kind == "mobile":
+            dims = ["latitude", "longitude"]
+        elif self.data_kind == "organized":
+            dims = ["time", "latitude", "longitude"]
+        else:
+            raise ValueError("invalid data_kind")
+
+        env_params = self.env_params
+
+        if self.data_kind == "mobile":
+            # compute_range needs indexing by ping_time
+            interp_plat = echodata.platform.interp(
+                {"location_time": echodata.platform["ping_time"]}
+            )
+
+            result = {}
+            for var, values in env_params.data_vars.items():
+                points = np.column_stack(
+                    (env_params["latitude"].data, env_params["longitude"].data)
+                )
+                values = values.data
+                xi = np.column_stack(
+                    (
+                        interp_plat["latitude"].data,
+                        interp_plat["longitude"].data,
+                    )
+                )
+                interp = scipy.interpolate.griddata(
+                    points, values, xi, method=self.interp_method
+                )
+                result[var] = ("ping_time", interp)
+            env_params = xr.Dataset(
+                data_vars=result, coords={"ping_time": interp_plat["ping_time"]}
+            )
+        else:
```

Comment:
Since "organized" is not supported for now, this is really just the "stationary" case, right?
If this is the case, the iterations like `for dim in dims` below would be redundant, since there is only 1 dimension `ping_time`.
Am I missing anything?
Could you clean up the code so that this intention is more explicit? or are you planning on reusing some of this for the "organized" case? down the line?

Reply:
Yes I was planning to use this for the `"organized"` case

Reply:
Just noticed this again -- could you add a TODO here so that the intention is clear?